permission/backend.go

```diff
@@ -0,0 +1,284 @@
+
+	contractBackEnd := ptype.ContractBackend{
+		EthClnt:    ethClnt,
+		Key:        key,
+		PermConfig: permConfig,
+		IsRaft:     isRaft,
+		UseDns:     useDns,
+	}
+
+	if permissionV2 {
+		return &v2.Init{
+			Backend: contractBackEnd,
+		}
+	}
+	return &v1.Init{
+		Backend: contractBackEnd,
+	}
+}
+
+func (p *PermissionCtrl) NewPermissionRoleService(txa ethapi.SendTxArgs) (ptype.RoleService, error) {
+	transactOpts, err := p.getTxParams(txa)
+	if err != nil {
+		return nil, err
+	}
+	return p.backend.GetRoleService(transactOpts, p.getContractBackend())
+}
+
+func (p *PermissionCtrl) NewPermissionOrgService(txa ethapi.SendTxArgs) (ptype.OrgService, error) {
+	transactOpts, err := p.getTxParams(txa)
+	if err != nil {
+		return nil, err
+	}
+	return p.backend.GetOrgService(transactOpts, p.getContractBackend())
+}
+
+func (p *PermissionCtrl) NewPermissionNodeService(txa ethapi.SendTxArgs) (ptype.NodeService, error) {
+	transactOpts, err := p.getTxParams(txa)
+	if err != nil {
+		return nil, err
+	}
+	return p.backend.GetNodeService(transactOpts, p.getContractBackend())
+}
+
+func (p *PermissionCtrl) NewPermissionAccountService(txa ethapi.SendTxArgs) (ptype.AccountService, error) {
+	transactOpts, err := p.getTxParams(txa)
+	if err != nil {
+		return nil, err
+	}
+	return p.backend.GetAccountService(transactOpts, p.getContractBackend())
+}
+
+func (p *PermissionCtrl) NewPermissionAuditService() (ptype.AuditService, error) {
+	return p.backend.GetAuditService(p.getContractBackend())
+}
+
+func (p *PermissionCtrl) NewPermissionControlService() (ptype.ControlService, error) {
+	return p.backend.GetControlService(p.getContractBackend())
+}
+
+func (p *PermissionCtrl) getContractBackend() ptype.ContractBackend {
+	return ptype.ContractBackend{EthClnt: p.ethClnt, Key: p.key, PermConfig: p.permConfig, IsRaft: p.isRaft, UseDns: p.isRaft}
+}
+
+func (p *PermissionCtrl) ConnectionAllowed(_enodeId, _ip string, _port, _raftPort uint16) (bool, error) {
+	cs, err := p.backend.GetControlService(p.getContractBackend())
+	if err != nil {
+		return false, err
+	}
+	return cs.ConnectionAllowed(_enodeId, _ip, _port, _raftPort)
+}
+
+func (p *PermissionCtrl) IsTransactionAllowed(_sender common.Address, _target common.Address, _value *big.Int, _gasPrice *big.Int, _gasLimit *big.Int, _payload []byte, transactionType core.TransactionType) error {
+	// If permissions model is not in use return nil
+	if core.PermissionModel == core.Default {
+		return nil
+	}
+
+	cs, err := p.backend.GetControlService(p.getContractBackend())
+	if err != nil {
+		return err
+	}
+
+	return cs.TransactionAllowed(_sender, _target, _value, _gasPrice, _gasLimit, _payload, transactionType)
+}
+
+func (p *PermissionCtrl) populateBackEnd() error {
+	backend := ptype.NewInterfaceBackend(p.node, false, p.dataDir)
+
+	switch p.permConfig.PermissionsModel {
+	case ptype.PERMISSION_V2:
+		p.backend = &v2.Backend{
+			Ib: *backend,
+		}
+		log.Debug("permission service: using v2 permissions model")
```

Comment:
It may be worthwhile logging this at `log.Info` level (the same applies for the v1 log message below).
Currently I don't think there's any Info level log which outputs the permission model being used. It's useful to have that when troubleshooting.

Reply:
Done